src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV2Writer.java

```diff
@@ -0,0 +1,165 @@
+package cuchaz.enigma.translation.mapping.serde;
+
+import com.google.common.base.Strings;
+import cuchaz.enigma.ProgressListener;
+import cuchaz.enigma.translation.mapping.EntryMap;
+import cuchaz.enigma.translation.mapping.EntryMapping;
+import cuchaz.enigma.translation.mapping.MappingDelta;
+import cuchaz.enigma.translation.mapping.MappingSaveParameters;
+import cuchaz.enigma.translation.mapping.tree.EntryTree;
+import cuchaz.enigma.translation.mapping.tree.EntryTreeNode;
+import cuchaz.enigma.translation.representation.entry.*;
+import cuchaz.enigma.utils.LFPrintWriter;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public final class TinyV2Writer implements MappingsWriter {
+
+	private static final String MINOR_VERSION = "0";
+	private final String obfHeader;
+	private final String deobfHeader;
+
+	public TinyV2Writer(String obfHeader, String deobfHeader) {
+		this.obfHeader = obfHeader;
+		this.deobfHeader = deobfHeader;
+	}
+
+	@Override
+	public void write(EntryTree<EntryMapping> mappings, MappingDelta<EntryMapping> delta, Path path, ProgressListener progress, MappingSaveParameters parameters) {
+		List<EntryTreeNode<EntryMapping>> classes =	StreamSupport.stream(mappings.spliterator(), false).filter(node -> node.getEntry() instanceof ClassEntry).collect(Collectors.toList());
+
+		try (PrintWriter writer = new LFPrintWriter(Files.newBufferedWriter(path))) {
+			writer.println("tiny\t2\t" + MINOR_VERSION + "\t" + obfHeader + "\t" + deobfHeader);
+
+			// no escape names
+
+			for (EntryTreeNode<EntryMapping> node : classes) {
+				writeClass(writer, node, mappings);
+			}
+		} catch (IOException ex) {
+			ex.printStackTrace(); // TODO add some better logging system
+		}
+	}
+
+	private void writeClass(PrintWriter writer, EntryTreeNode<EntryMapping> node, EntryMap<EntryMapping> tree) {
+		writer.print("c\t");
+		ClassEntry classEntry = (ClassEntry) node.getEntry();
+		String fullName = classEntry.getFullName();
+		writer.print(fullName);
+		Deque<String> parts = new LinkedList<>();
+		do {
+			EntryMapping mapping = tree.get(classEntry);
+			if (mapping != null) {
+				parts.addFirst(mapping.getTargetName());
+			} else {
+				parts.addFirst(classEntry.getName());
+			}
+			classEntry = classEntry.getOuterClass();
+		} while (classEntry != null);
+
+		String mappedName = String.join("$", parts);
+
+		writer.print("\t");
+		if (!fullName.equals(mappedName)) {
```

Comment:
This line costed me 3 hours of debugging :D
If it's the same it doesn't mean it's not mapped!